lib/matplotlib/axes/_axes.py

```diff
@@ -1241,6 +1245,7 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         return colls
 
     #### Basic plotting
+    @unpack_labeled_data()
```

Comment:
this has an implicit `label_namer="y"`, which results in an label if someone passes in `y="col", data=data` or `y=df["col"]` (e.g. it must be a kwarg). Is that ok?